Tests/js/views/services/assets/ez-discoverybarviewservice-tests.js

```diff
@@ -88,7 +89,147 @@ YUI.add('ez-discoverybarviewservice-tests', function (Y) {
+                        value: this.location
+                    }
+                }
+            });
+
+            this.activeView = new Y.eZ.LocationViewView();
+            this.app = new Mock();
+            this.capi = new Mock();
+
+            Mock.expect(this.app, {
+                method: 'get',
+                args: ["activeView"],
+                returns: this.activeView
+            });
+
+            this.service = new Y.eZ.DiscoveryBarViewService({
+                app: this.app,
+                capi: this.capi,
+            });
+        },
+
+        tearDown: function () {
+            delete Y.eZ.LocationViewView;
+            delete Y.eZ.Location;
+            this.service.destroy();
+            delete this.service;
+            delete this.app;
+            delete this.capi;
+        },
+
+        "Should fire contentDiscover with a starting location on treeAction event": function () {
+            var contentDiscoverFired = false;
+
+            this.service.on('contentDiscover', Y.bind(function (e) {
+                contentDiscoverFired = true;
+
+                Assert.areSame(
+                    e.config.startingLocationId, this.locationId,
+                    "startingLocationId should have the id of the current location of the locationViewView"
+                );
+            }, this));
+            this.service.fire('findAction');
+
+            Assert.isTrue(contentDiscoverFired, 'contentDiscover should be fired');
+        },
+
+        "Should fire contentDiscover without starting location when current view is NOT a locationViewView": function () {
+            var contentDiscoverFired = false;
+
+            this.activeView = new Y.Base();
+
+            Mock.expect(this.app, {
+                method: 'get',
+                args: ["activeView"],
+                returns: this.activeView
+            });
+            this.service.on('contentDiscover', Y.bind(function (e) {
+                contentDiscoverFired = true;
+
+                Assert.isUndefined(
+                    e.config.startingLocationId,
+                    "startingLocationId should be undefined"
+                );
+            }, this));
+            this.service.fire('findAction');
+
+            Assert.isTrue(contentDiscoverFired, 'contentDiscover should be fired');
+        },
+
+        "Should navigate to the selected location when content is discovered": function () {
+            var locationMock = new Y.Mock(),
+                contentMock = new Y.Mock(),
+                fakeEventFacade = {selection: {location: locationMock, content: contentMock}};
+
+            Mock.expect(this.app, {
+                method: 'get',
+                args: ["activeView"],
+                callCount: 2,
+                returns: this.activeView
+            });
+
+            Y.Mock.expect(locationMock, {
+                method: 'get',
+                args: ['id'],
+                returns: this.locationId
```

Comment:
you use the same location id for both the starting Location and the picked Location